langpuppet/gapi.go

```diff
@@ -0,0 +1,327 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package langpuppet
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"sync"
+
+	multierr "github.com/hashicorp/go-multierror"
+	errwrap "github.com/pkg/errors"
+
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/gapi"
+	"github.com/purpleidea/mgmt/pgraph"
+
+	"github.com/purpleidea/mgmt/lang"
+	"github.com/purpleidea/mgmt/puppet"
+
+	"github.com/urfave/cli"
+)
+
+const (
+	// Name is the name of this frontend.
+	Name       = "langpuppet"
+	FlagPrefix = "lp-"
+)
+
+func init() {
+	gapi.Register(Name, func() gapi.GAPI { return &GAPI{} }) // register
+}
+
+// GAPI implements the main langpuppet GAPI interface.
```

Comment:
If you could extend this comment with a short description explaining the magic, that would be great.